File: src/eds-transforms/src/main/java/org/endeavourhealth/transform/ui/models/UIProblem.java
package org.endeavourhealth.transform.ui.models;

public class UIProblem extends UICondition {
    private UICode significance;

    public UICode getSignificance() {
        return significance;
    }

    public UIProblem setSignificance(UICode significance) {
        this.significance = significance;
        return this;
    }
}

File: src/eds-ui/src/main/frontend/app/recordViewer/models/resources/clinical/UIProblem.ts
import {UIPractitioner} from "../admin/UIPractitioner";
import {UICode} from "../../types/UICode";
import {UICondition} from "./UICondition";
import {UIDate} from "../../types/UIDate";

export class UIProblem extends UICondition {
    expectedDuration: number;
    lastReviewDate: UIDate;
    lastReviewer: UIPractitioner;
    significance: UICode;
    relatedProblem: UIProblem;
    relationshipType: string;
}
